core/src/com/unciv/logic/GameInfo.kt

```diff
@@ -292,6 +294,55 @@ class GameInfo {
         }
     }
 
+    fun notifyExploredResources(civInfo: CivilizationInfo, resourceName: String, maxDistance: Int, showForeign: Boolean) {
+        // Calling with `maxDistance = 0` removes distance limitation.
+        data class CityTileAndDistance(val city: CityInfo, val tile: TileInfo, val distance: Int)
+
+        var exploredRevealTiles:Collection<TileInfo>
```

Comment:
Since you're using so many mapping functions, this should definitely be a sequence

Reply:
Lazy evaluation? Cool. Honestly, I looked up the documentation for `Collection`s, but I kinda assumed `Sequence`s must just be a synonym or some sort of ABC for `List`s.